R/get-majority-t1d-diagnoses.R

```diff
@@ -0,0 +1,42 @@
+#' Determine Majority of Type 1 Diabetes Diagnoses
+#'
+#' This function evaluates whether an individual has a majority of type 1 diabetes-specific hospital diagnoses (DE10) among all type-specific diabetes primary diagnoses (DE10 & DE11) from medical departments.
+#' If an individual has any type-specific diabetes diagnoses from endocrinology departments, the majority is determined only among these contacts.
+#'
+#' @param n_t1d_endocrinology Numeric vector. The count of type 1 diabetes primary diagnoses from endocrinology departments.
+#' @param n_t2d_endocrinology Numeric vector. The count of type 2 diabetes primary diagnoses from endocrinology departments.
+#' @param n_t1d_medical Numeric vector. The count of type 1 diabetes primary diagnoses from other medical departments.
+#' @param n_t2d_medical Numeric vector. The count of type 2 diabetes primary diagnoses from other medical departments.
+#'
+#' @return A logical vector indicating whether the majority of primary diagnoses are related to type 1 diabetes.
+#' @keywords internal
+#'
+#' @examples
+#' # Example usage within the include_diabetes_diagnoses() pipeline:
+#' include_diabetes_diagnoses(lpr2, lpr3) |>
+#'  dplyr::mutate(
+#'    majority_t1d_primary_diagnoses = get_majority_of_t1d_diagnoses(
+#'      n_t1d_endocrinology,
+#'      n_t2d_endocrinology,
+#'      n_t1d_medical,
+#'      n_t2d_medical
+#'    )
+#'  ) |>
+#'  dplyr::arrange(.data$pnr)
+#'
+#' @seealso [dplyr::case_when()] for condition handling.
+get_majority_of_t1d_diagnoses <- function(n_t1d_endocrinology,
+                                          n_t2d_endocrinology,
+                                          n_t1d_medical,
+                                          n_t2d_medical) {
+
+  dplyr::case_when(
+    sum(n_t1d_endocrinology, n_t2d_endocrinology, na.rm = TRUE) > 0 ~
+      dplyr::coalesce(n_t1d_endocrinology, 0) > dplyr::coalesce(n_t2d_endocrinology, 0),
+    # coalesce() is used to handle NAs when evaluating the right-hand side of the expression by returning those as zero
```

Comment:
Rather than use coalesce, let's make sure there are no NAs before putting it into this function.